deckhand/engine/schema/v1_0/default_schema.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+substitution_schema = {
+    'type': 'object',
+    'properties': {
+        'dest': {'type': 'string'},
+        'src': {
+            'type': 'object',
+            'properties': {
+                'apiVersion': {
+                    'type': 'string',
+                    'choices': ['deckhand/v1']
+                },
+                'kind': {'type': 'string'},
+                'name': {'type': 'string'}
+            },
+            'additionalProperties': False,
+            'required': ['apiVersion', 'kind', 'name']
+        }
+    },
+    'additionalProperties': False,
+    'required': ['dest', 'src']
+}
+
+schema = {
```

Comment:
I don't think you will need to do particularly manual validation of the data sections.  The intention is for the schemas of the data section to be specified at runtime via documents either on the filesystem or through api endpoints.

It's not 100% decided whether we will inject these from a centralized place of ownership or whether the individual services will own particular schemas.

There are definitely additional kinds of validation that will not be easily expressed by schemas as you begin implementing layering and substitution (e.g. does the source path exist in the source document), but even then the final validation of a concrete document should be possible through a provided schema.